h/services/bulk_api/group.py

```diff
@@ -0,0 +1,39 @@
+from dataclasses import dataclass
+from typing import List
+
+import sqlalchemy as sa
+from sqlalchemy.orm import Session
+
+from h.models import Annotation, Group
+from h.services.bulk_api.core import date_match
+
+
+@dataclass
+class BulkGroup:
+    authority_provided_id: str
+
+
+class BulkGroupService:
+    """A service for retrieving groups in bulk."""
+
+    def __init__(self, db: Session):
+        self._db = db
+
+    def group_search(
+        self, groups: List[str], annotations_created: dict
+    ) -> List[BulkGroup]:
+        query = (
+            sa.select([Group.authority_provided_id])
+            .join(Annotation, Group.pubid == Annotation.groupid)
```

Comment:
Joining on the main annotation table. Once AnnotationSlim contains all annotations we could switch to that.